### DISCUSSION TITLE
Articles Ovewrites and Alternative Read more

### DISCUSSION THREAD

**Discussion by niosme** [joomla/joomla-cms#47034] [2. Cold Features]
### Steps to reproduce the issue
We create an ovewrite e.g. products for articles.

There is ovewrite on menu type blog layout for their articles so we can say every article inside this will have the products ovewrite.

But if we assign a new menu item to a signle article the ovewrite doesnt work because its ovewritten by the new menu item for a reasong. But on a single artcile menu type there is no option to select the ovewrite layout. We should go and change it from inside the article.

So the blog menu item if we create another menu item of a single article inside the blog menu this article will no longer use the ovewrite. The single menu item ovewrites the functionallity. But there is no option also from the single menu item to change the layout but only from inside the article directly.

Another thing only exists inside the article is the alternative_readmore which we cannot handle it from category or from menu item and we have to set it from every signle article.
E.g. if we want to set it in a category from the menu item blog layout or from the category of the blog com_content> categories.
but i see there is option to handle it globally also but no option in the backend.

### Expected result
The blog menu handles its ovwrite layout functionallity or handle also the layout from the single article menu item type

Handle alternative readmore from each category or from each menu type blog layout.

### Actual result

We have to directly set the layout from inside the article else no ovewrites are being used.

### System information (as much as possible)

Joomla 5.1.2 but its happening on every joomla.

**Comment by chmst** [comment]
> But if we assign a new menu item to a signle article the ovewrite doesnt work because its ovewritten by the new menu item for a reasong. But on a single artcile menu type there is no option to select the ovewrite layout. We should go and change it from inside the article.

If you have a menu item for a single article, Then you do not need a readmore Button. Single articles always are displayed in full length without read more button. 

Maybe I don't understan correctly what you mean - or do you have special overrides for your single article?

**Comment by niosme** [comment]
> > But if we assign a new menu item to a signle article the ovewrite doesnt work because its ovewritten by the new menu item for a reasong. But on a single artcile menu type there is no option to select the ovewrite layout. We should go and change it from inside the article.
> 
> If you have a menu item for a single article, Then you do not need a readmore Button. Single articles always are displayed in full length without read more button.
> 
> Maybe I don't understan correctly what you mean - or do you have special overrides for your single article?

Issue 1:
i was just talking about when we do an ovewrite on our templates/cassiopeia/html/com_content/article/myovewrite.php
There is a conflict when we are setting a blog menu with articles and we set a layout ovewrite for our blog articles.
If another menu item is assigned directly to the article the above blog menu layout no longer exists for it.

Issue 2:
The read more is another thing that we only be able to change it inside an article but not for a whole category of articles.i mean the read more as a word in the options of the article.We could change it also from the category or the blog menu item also.
